runFetchDataAnalyseNew.py

```diff
@@ -0,0 +1,202 @@
+import matplotlib.pyplot as plt
+from shapely.ops import split
+from shapely import get_coordinates
+from shapely.geometry import Point, LineString
+import matplotlib.patheffects as pe
+import seaborn as sns
+import math
+from sklearn.metrics import r2_score, mean_squared_error
+
+
+from avaframe.in3Utils import cfgUtils
+from avaframe.in3Utils import logUtils
+import avaframe.in3Utils.fileHandlerUtils as fU
+import avaframe.in3Utils.geoTrans as gT
+import avaframe.out3Plot.amaPlots as aP
+import avaframe.in2Trans.ascUtils as IOf
+import DFAPathGeneration as DFAPath
+import avaframe.out3Plot.plotUtils as pU
+
+# use latex in str
+# plt.rcParams["text.usetex"] = True
+# +++++++++SETUP CONFIGURATION++++++++++++++++++++++++
+# log file name; leave empty to use default runLog.log
+logName = 'runFetchDataAnalyse'
+
+# Load avalanche directory and accessFile path from general configuration file
+dirPath = pathlib.Path(__file__).parents[0]
+cfgMain = cfgUtils.getGeneralConfig(nameFile=(dirPath / 'amaConnectorCfg.ini'))
+avalancheDir = pathlib.Path(cfgMain['MAIN']['avalancheDir'])
+fU.makeADir(avalancheDir)
+accessfile = pathlib.Path(cfgMain['MAIN']['accessFile'])
+queryString = cfgMain['MAIN']['queryString']
+
+# load info on setup for analysis
+nonEmptyCols = cfgMain['FILTERING']['nonEmptyAttributes'].split('|')
+addAttributes = cfgMain['FILTERING']['addAttributes'].split('|')
+resampleDist =  cfgMain['FILTERING'].getfloat('resampleDist')
+
+# Start logging
+log = logUtils.initiateLogger(avalancheDir, logName, modelInfo='AmaConnector')
+log.info('MAIN SCRIPT')
+log.info('Current search: %s', avalancheDir)
+
+# fetch all info of the database according to queryString and return dataFrame
+dbData = gD.grabAllComplete(avalancheDir, queryString=queryString, accessfile=accessfile)
+log.info('Fetched %d entries from data base ' % (len(dbData)))
+for index, row in dbData.iterrows():
+    log.info('%s, event id: %s, index %s' % (row['path_name'], row['event_id'], index))
+
+
+
+# filter db according to nonEmptyCols and convert geometry entries to desired projection
+# resample thalweg for higher resolution
+dbFiltered = aU.fetchGeometryInfo(dbData, 'epsg:4326', cfgMain['MAIN']['projstr'],
+    cfgMain['FILTERING']['geomStr'], nonEmptyCols, addAttributes, resampleDist)
+
+log.info('Filtered db data and converted to %s' % cfgMain['MAIN']['projstr'])
+log.info('Events found for path name:')
+for index, row in dbFiltered.iterrows():
+    log.info('%s, event id: %s' % (row['path_name'], row['event_id']))
+
+#Import settings from .ini-File
+projstr = cfgMain['MAIN']['projstr']
+resDist = cfgMain['MAIN'].getfloat('resamplePathFit')
+dsMin = cfgMain['MAIN'].getfloat('dsMin')
+slope1 = cfgMain['MAIN'].getfloat('slopeAngle1')
+slope2 = cfgMain['MAIN'].getfloat('slopeAngle2')
+
+mask = dbFiltered.duplicated(subset='path_id', keep ='first')
+dbFiltered = dbFiltered[~mask]
+
+dbFiltered.dropna(subset=['geom_rel_event_pt3d_epsg:31287'], inplace=True)
```

Comment:
drop entries with nan values for rel event point - this can be handled in the filtering that is done above for nonEmptyCols in ini file - so remove this here?

Reply:
removed